test-website/readme.md

```diff
@@ -0,0 +1,145 @@
+  3. Add the following URLs to the Visual Editor settings (on the settings page)
+
+     - `http://localhost:3000`
+     - `http://localhost:3000/blog`
+     - `http://localhost:3000/blog/why-steampunk-rabbits-are-the-future-of-work`
+
+       > [!NOTE]  
+       > The last two URLs are for testing purposes only. You do not need to add every page URL of your website to
+       > access them with the Visual Editor.
+
+  4. Open the Visual Editor Module
+
+## Test Modes
+
+Use the following commands to run predefined test environments. Find the corresponding code by searching for
+`useVisualEditingTest` or `testCase` in the test-website directory to see how it is set up.
+
+> [!NOTE]  
+> Be sure to stop other servers running on localhost:3000
+
+```sh
+# commands
+
+pnpm visual-editing:monolith
+
+pnpm visual-editing:ssr
+pnpm visual-editing:ssr--refresh-all
+pnpm visual-editing:ssr--refresh
+pnpm visual-editing:ssr--refresh-customized # http://localhost:3000/blog
+pnpm visual-editing:ssr--methods # http://localhost:3000
+
+pnpm visual-editing:dev
+pnpm visual-editing:dev--refresh-all
+pnpm visual-editing:dev--refresh
+pnpm visual-editing:dev--refresh-customized # http://localhost:3000/blog
+pnpm visual-editing:dev--methods # http://localhost:3000
+
+pnpm visual-editing:ssg # http://localhost:3000/blog/why-steampunk-rabbits-are-the-future-of-work
+```
+
+### Rendering modes:
+
+- Monolith / server only rendering: use as with php
+- SSG: live data only on hydration
+- SSR: always with live data
+- Dev mode: if you want to play around
+
+### Test Cases
+
+- `basic`
+  - No `onSaved` callback — the web page inside the iframe will be reloaded (`window.location.reload()`)
+  - Editable elements are added and removed globally on route change
+  - Rendering modes: `monolith` || `dev` || `ssr`
+  - Search for `testCase === 'basic'`
+- `refresh-all`
+  - `onSaved` will refetch all data with `refreshNuxtData()`
+  - Editable elements will be added and removed globally on route changes
+  - Rendering modes: `dev` || `ssr`
+  - Search for `testCase === 'refresh-all'`
+- `refresh` _(smooth, recommended)_
+  - `onSaved` will call refetch all data where it was originally fetched.
+  - Editable elements are only removed globally on route changes, they are added in nested components where data is
+    fetched, so that their `refresh()` function can be passed to `onSaved`. This should provide the smoothest experience
+    without layout shifts.
+  - Rendering modes: `dev` || `ssr`
+  - Search for `testCase === 'refresh'`
+- `refresh-customized`
+  - The best way to see this on the test website is to open this page in the Visual Editor: http://localhost:3000/blog
+  - Exactly the same as `refresh` above, except that custom classes are attached to some editable elements to
+    demonstrate customizability.
+  - Rendering modes: `dev` || `ssr`
+  - Search for `testCase === 'refresh-customized'`
+- `refresh-methods`
```

Comment:
Possibly confusing given that the associated command above is simply `pnpm visual-editing:ssr--methods`  rather than `pnpm visual-editing:ssr--refresh-methods`